packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,267 @@
+import {
+	CCCommand,
+	CCCommandOptions,
+	ccValue,
+	ccValueMetadata,
+	CommandClass,
+	commandClass,
+	CommandClassDeserializationOptions,
+	expectedCCResponse,
+	gotDeserializationOptions,
+	implementedVersion,
+} from "./CommandClass";
+// All the supported commands
+export enum BarrierOperatorCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+	CapabilitiesGet = 0x04,
+	CapabilitiesReport = 0x05,
+	EventSet = 0x06,
+	EventGet = 0x07,
+	EventReport = 0x08,
+}
+
+/**
+ * @publicAPI
+ */
+export enum BarrierState {
+	Closed = 0x00,
+	Closing = 0xfc,
+	Stopped = 0xfd,
+	Opening = 0xfe,
+	Open = 0xff,
+}
+
+// @publicAPI
+export enum SignalType {
+	Audible = 0x01,
+	Visual = 0x02,
+}
+
+// @publicAPI
+export enum SignalState {
+	OFF = 0x00,
+	ON = 0xff,
+}
+
+@commandClass(CommandClasses["Barrier Operator"])
+@implementedVersion(1)
+export class BarrierOperatorCC extends CommandClass {
+	declare ccCommand: BarrierOperatorCommand;
+}
+
+interface BarrierOperatorCCSetOptions extends CCCommandOptions {
+	targetValue: boolean;
+}
+
+@CCCommand(BarrierOperatorCommand.Set)
+export class BarrierOperatorCCSet extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options:
+			| CommandClassDeserializationOptions
+			| BarrierOperatorCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			this.targetValue = options.targetValue;
+		}
+	}
+
+	public targetValue: boolean;
+
+	public serialize(): Buffer {
+		const payload: number[] = [this.targetValue ? 0xff : 0x00];
+		this.payload = Buffer.from(payload);
+		return super.serialize();
+	}
+
+	public toLogEntry(): MessageOrCCLogEntry {
+		return {
+			...super.toLogEntry(),
+			message: { "target value": this.targetValue },
+		};
+	}
+}
+
+@CCCommand(BarrierOperatorCommand.Report)
+export class BarrierOperatorCCReport extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 1);
+
+		/* 
+		  return values state and position value
+		  if state is 0 - 99 or FF (100%) return the appropriate values.
+		  if state is different just use the table and 
+		  return undefined position
+		*/
+		const payloadValue = this.payload[0];
+		if ((payloadValue >= 0 && payloadValue <= 99) || payloadValue == 255) {
+			// convert position % to 100, in case of position is Open
+			const payloadPosition = payloadValue === 255 ? 100 : payloadValue;
+			this._barrierPosition = payloadPosition;
+			if (payloadValue > 0 && payloadValue <= 99) {
+				this._barrierState = undefined;
+			} else {
+				this._barrierState = payloadValue;
+			}
+		} else {
+			this._barrierPosition = undefined;
+			this._barrierState = payloadValue;
+		}
```

Comment:
This could probably be simplified logic-wise but let's keep it like this for now.

Reply:
I simplified the logic on my next commit